static/docs/commands-reference/install.md

```diff
@@ -23,13 +50,204 @@ Install dvc hooks into the repository
+```
+
+Second, let's install the requirements. But before we do that, we **strongly**
+recommend creating a virtual environment with `virtualenv` or a similar tool:
+
+```dvc
+    $ cd example-get-started
+    $ virtualenv -p python3 .env
+    $ source .env/bin/activate
+```
+
+Now, we can install requirements for the project:
+
+```dvc
+    $ pip install -r requirements.txt
+```
+
+Then download the precomputed data using:
+
+```dvc
+    $ dvc pull --all-branches --all-tags
+```
+
+This data will be retrieved from a preconfigured remote cache.
+
+</details>
+
+## Example: Checkout both DVC and SCM
+
+Let's start our exploration with the impact of `dvc install` on the
+`dvc checkout` command.  Remember that switching from one SCM tag or branch to
+another changes the set of DVC files in the workspace, which then also changes
+the data files that should be in the workspace.
+
+With the Getting Started example workspace described above, let's first list
+the available tags:
+
+```dvc
+    $ git tag
+
+    0-empty
+    1-initialize
+    2-remote
+    3-add-file
+    4-sources
+    5-preparation
+    6-featurization
+    7-train
+    8-evaluation
+    9-bigrams
+    baseline-experiment
+    bigrams-experiment
+```
+
+These tags are used to mark points in the development of this workspace, and to
+document specific experiments conducted in the workspace.  To take a look at one
+we check-out the workspace using the SCM (in this case Git):
+
+```dvc
+    $ git checkout 6-featurization
+    Note: checking out '6-featurization'.
+
+    You are in 'detached HEAD' state.  ...
+
+    $ dvc status
+
+    featurize.dvc:
+        changed outs:
+            modified:           data/features
+
+    $ dvc checkout
+
+    [##############################] 100% Checkout finished!
+
+    $ dvc status
+
+    Pipeline is up to date. Nothing to reproduce.
+```
+
+After running `git checkout` we are also shown a message saying _You are in
+'detached HEAD' state_, and the Git documentation explains what that means.
+Bottom line is returning the workspace to a normal state requires the
+command `git checkout master`.
+
+We also see that `dvc status` tells us about differences between the workspace
+and the data files currently in the workspace. Git changed the DVC files in
+the workspace, which changed references to data files. What `dvc status` did is
+inform us the data files in the workspace no longer matched the checksums in
+the DVC files. Running `dvc checkout` then checks out the corresponding data
+files, and now `dvc status` tells us the data files match the DVC files.
+
+Now let's see how this changes after running `dvc install`
+
+```dvc
+    $ git checkout master
+    Previous HEAD position was d13ba9a add featurization stage
+    Switched to branch 'master'
+    Your branch is up to date with 'origin/master'.
+
+    $ dvc checkout
+    [##############################] 100% Checkout finished!
+```
+
+Start by resetting the workspace to he at the head commit.
```

Comment:
Should we move this before the code? I was misled a bit and was expecting `dvc install` to appear before.

Reply:
Did some rewording.